std/prettier/main.ts

```diff
@@ -343,10 +356,134 @@ async function* getTargetFiles(
   }
 }
 
+/**
+ * auto detect prettier configuration file and return config if file exist.
+ */
+async function autoResolveConfig(): Promise<PrettierBuildInOptions> {
+  const configFileNamesMap = {
+    ".prettierrc.json": 1,
+    ".prettierrc.yaml": 1,
+    ".prettierrc.yml": 1,
+    ".prettierrc.js": 1,
+    ".prettierrc.ts": 1,
+    "prettier.config.js": 1,
+    "prettier.config.ts": 1,
+    ".prettierrc.toml": 1
+  };
+
+  const files = await Deno.readDir(".");
+
+  for (const f of files) {
+    if (f.isFile() && configFileNamesMap[f.name]) {
+      const c = await resolveConfig(f.name);
+      if (c) {
+        return c;
+      }
+    }
+  }
+
+  return;
+}
+
+/**
+ * parse prettier configuration file.
+ * @param filepath the configuration file path.
+ *                 support extension name with .json/.toml/.js
+ */
+async function resolveConfig(
+  filepath: string
+): Promise<PrettierBuildInOptions> {
+  let config: PrettierOptions = undefined;
+
+  function generateError(msg: string): Error {
+    return new Error(`Invalid prettier configuration file: ${msg}.`);
+  }
+
+  const raw = new TextDecoder().decode(await Deno.readFile(filepath));
+
+  switch (path.extname(filepath)) {
+    case ".json":
+      try {
+        config = JSON.parse(raw) as PrettierOptions;
+      } catch (err) {
+        throw generateError(err.message);
+      }
+      break;
+    case ".yml":
+    case ".yaml":
+      // TODO: Unimplemented loading yaml / yml configuration file yet.
+      break;
+    case ".toml":
+      try {
+        config = toml.parse(raw) as PrettierOptions;
```

Comment:
If prettier already has a toml config file - does that mean we can give them the toml file without parsing it?

Reply:
sorry, I don't quite understand what you means.

Are you saying that we don't have to parse the toml file?

Reply:
Yes that's what I'm wondering. Like is there already code for TOML parsing somewhere inside of https://github.com/denoland/deno/tree/master/std/prettier/vendor ? I don't know.

Reply:
It would be preferable to not have to parse prettier files, if they already do that... 

(It would be better if there was only one TOML parser in std tho, and not two.)

Reply:
https://github.com/denoland/deno/blob/26bf928d28496920bb86934b06ac4342a0998597/std/prettier/vendor/index.d.ts#L260-L274

Prettier does have this API.

But reading the file is the nodejs API.

Currently, Deno not compatible with the nodejs API.

Reply:
I see. Thanks for the explanation. I don't like that these prettier bundles are super huge and probably contain a lot of redundant code... but I don't see any easy way of resolving that now.